cloud-azure/src/main/resources/templates/arm-v2.ftl

```diff
@@ -255,13 +255,14 @@
                        "[concat('Microsoft.Network/virtualNetworks/', parameters('virtualNetworkNamePrefix'))]"
                        </#if>
                        <#if loadBalancerMapping[instance.groupName]?? && (loadBalancerMapping[instance.groupName]?size > 0)>
+                           <#if !noFirewallRules || !noPublicIp || !existingVPC>,</#if>
```

Comment:
this should be <#if !existingVPC && (!noFirewallRules || !noPublicIp) > instead

Reply:
I think you're right that my condition is wrong, but I think it's wrong in a different way.

Here's a reformatted collection of the tags:
```
<#if !noFirewallRules>
   <#if !isUpscale && (! securityGroups[instance.groupName]?? || ! securityGroups[instance.groupName]?has_content)>
       "[concat('Microsoft.Network/networkSecurityGroups/', variables('${instance.groupName?replace('_', '')}secGroupName'))]"
   </#if>
</#if>

<#if !noPublicIp>
   <#if !noFirewallRules>
       <#if !isUpscale && (! securityGroups[instance.groupName]?? || ! securityGroups[instance.groupName]?has_content)>
           ,
       </#if>
   </#if>
   "[concat('Microsoft.Network/publicIPAddresses/', parameters('publicIPNamePrefix'), '${instance.instanceId}')]"
</#if>

<#if !existingVPC>
   <#if !noFirewallRules || !noPublicIp>
       ,
   </#if>
"[concat('Microsoft.Network/virtualNetworks/', parameters('virtualNetworkNamePrefix'))]"
</#if>

<#if loadBalancerMapping[instance.groupName]?? && (loadBalancerMapping[instance.groupName]?size > 0)>
   <#if !noFirewallRules || !noPublicIp || !existingVPC>,</#if>
   <#list loadBalancerMapping[instance.groupName] as loadBalancer>
       <#--
          we have to define a dependency between the NIC and the address pool it belongs to
          this makes every instance in the gateway group depend on every load balancer address pool
          when we add support for multiple load balancers, this will have to be updated.
       -->
       "[resourceId('Microsoft.Network/loadBalancers', '${loadBalancer.name}')]"<#sep>,</#sep>
   </#list>
</#if>
```
The general pattern is that we have an `if` wrapping the dependency we want to add, then an inner if wrapping the placement of a comma to separate it from preceding deps.

There are essentially 4 dependency clauses:
* networkSecurityGroup
* publicIpAddress
* virtualNetworks
* loadBalancers

The NSG is present if:
```
<#if !noFirewallRules>
   <#if !isUpscale && (! securityGroups[instance.groupName]?? || ! securityGroups[instance.groupName]?has_content)>
```
So, I think I'm missing that condition from my `if` -- I only have the `!noFirewallRules`, when really I need to know that the second condition is also true.

Any single one of `!noPublicIp`, `!existingVPC`, or `!noFirewallRules && (!isUpscale && (! securityGroups[instance.groupName]?? || ! securityGroups[instance.groupName]?has_content))` will result in there being a dependency _before_ the load balancers, so I should `or` those three conditions.

Reply:
@pdarvasi I noticed that we're always passing `noFirewallRules` as `false` from the `AzureTemplateBuilder` class. Is that the only class that uses the `arm-v2.ftl` template? If so, we can simplify these conditions a bunch.

Reply:
Yes, ideally, `noFirewallRules` should be eliminated.

Reply:
@brycederriso are you going to make those simplifications?

Reply:
@pdarvasi got to the changes before me, thanks!

Assign is a useful directive, thanks for showing how it's used!
```
                  <#assign createAndNoSecGroup = !isUpscale && (! securityGroups[instance.groupName]?? || ! securityGroups[instance.groupName]?has_content)>
```